CredentialProvider.Microsoft/CredentialProviders/Vsts/VstsBuildTaskMsalTokenProvidersFactory.cs

```diff
@@ -0,0 +1,38 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.Artifacts.Authentication;
+using Microsoft.Extensions.Logging;
+using NuGetCredentialProvider.Util;
+
+namespace NuGetCredentialProvider.CredentialProviders.Vsts;
+
+internal class VstsBuildTaskMsalTokenProvidersFactory : ITokenProvidersFactory
```

Comment:
Move this class under the VstsBuildTaskServiceEndpoint folder where it's used

Reply:
yes